qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1303 @@
+      self.walletList.setMaximumHeight(200)
+      # Avoid style hacks that can blank the viewport on some Qt builds
+
+      # Add wallet list to grid (no title needed)
+      walletGrid.addWidget(self.walletList, 0, 0)
+      # Remove row stretch to prevent excessive expansion
+      walletGrid.setRowStretch(0, 0)
+
+      # Add grid to wallet frame
+      walletFrameLayout.addLayout(walletGrid)
+
+      # Wallet list auto-refreshes when bridge is ready
+
+      # Add wallet frame to main layout
+      mainLayout.addWidget(walletFrame)
+      mainLayout.addStretch()
+
+      tab.setLayout(mainLayout)
+      return tab
+
+   def refreshWalletList(self):
+      """Refresh wallet list display - required by WalletMap for migration."""
+      # This method is called by WalletMap.migrateWallet after migration
+      # CRITICAL: May be called from bridge thread, use TheSignalExecution
+      # to ensure UI updates happen in the main thread
+      TheSignalExecution.executeMethod(self.loadWalletList)
+
+   def getWalletManager(self):
+      """Return the wallet manager for transfer to main window."""
+      return self.walletManager
+
+   def logEntropy(self):
+      """Log entropy for cryptographic randomness - required by migration."""
+      self.entropyAccum.write(pack('d', RightNow()))
+      self.entropyAccum.write(pack('i', QtGui.QCursor.pos().x()))
+      self.entropyAccum.write(pack('i', QtGui.QCursor.pos().y()))
+
+   def registerWidgetActivateTime(self, widget):
+      """Register widget for entropy accumulation - required by migration."""
+      # This patches widget event handlers to log entropy from user interactions
+      # Required for DlgWalletMigration's SetPassphraseFrame
+      setupManager = self
+
+      def newKPE(wself, event=None):
+         setupManager.logEntropy()
+         super(wself.__class__, wself).keyPressEvent(event)
+
+      def newKRE(wself, event=None):
+         setupManager.logEntropy()
+         super(wself.__class__, wself).keyReleaseEvent(event)
+
+      def newMPE(wself, event=None):
+         setupManager.logEntropy()
+         super(wself.__class__, wself).mousePressEvent(event)
+
+      def newMRE(wself, event=None):
+         setupManager.logEntropy()
+         super(wself.__class__, wself).mouseReleaseEvent(event)
+
+      widget.keyPressEvent     = MethodType(newKPE, widget)
+      widget.keyReleaseEvent   = MethodType(newKRE, widget)
+      widget.mousePressEvent   = MethodType(newMPE, widget)
+      widget.mouseReleaseEvent = MethodType(newMRE, widget)
+
+   def loadWalletList(self):
+      """Query bridge for wallets and render actions per type/state."""
+      self.walletList.clear()
+      self.walletIdToCheckbox.clear()
+      try:
+         addedAny = False
+         # Single bridge query; if it raises, show placeholder and return
+         try:
+            wltList = TheBridge.wltManager.listWallets()
+         except Exception as e:
+            addPlaceholderRow(
```

Comment:
Can this even happen? bridge should be ready before the dialog spawns, by virtue of event driven model